### DISCUSSION TITLE
Lost SoC and other FoxESS Inverter Issues

### DISCUSSION THREAD

**Discussion by spmaid** [StealthChesnut/HA-FoxESS-Modbus#58] [General]
Sorry for posting this here, but I don't use Facebook and so don't have access to the group on there.  Yesterday there was mention about the issue with 'lost' SoC overnight, and also the forced charging issue (which is the self protect kicking in).  I followed up with FoxESS support yesterday and they got back overnight to confirm they do now see the issue and expect to have a firmware fix coming soon, hopefully by end of the month or early next.

I am still questioning with them why the self protect charging is kicking in (for me) at 19%.  They are saying sometimes the SoC might not be accurate etc. but I have challenged them on that as I don't see that is acceptable that I can no longer risk going below 20% SoC, it used to happily go down to 10% and as we get less and less solar now, that extra 10% will be very much needed. 

Happy to post here again when I hear any more.

**Comment by tagware** [comment]
Well done for reporting the bug.

As I have noticed the same behaviour.

**Comment by C00K50N** [comment]
For anyone able to understand code better than me...

In order to monitor the actual SoC, or at least provide a measure against the stated SoC to see if it's far out, I was wondering if we'd be able to create a sensor to do the following...

Monitor sensor.battery_soc to see when it hits 100% (so it would recalibrate each time it's full).

If SoC is greater than or equal to 99%
Reset a counter to a specified value [user's battery capacity in kWh, e.g. 15.6]
Monitor sensor.bat_discharge_sum and sensor.bat_charge_sum from that point to get a measured remaining charge in kWh not %
Divide measured remaining charge by max capacity to give a % figure to compare against the stated SoC
Counter should reset next time battery is at 99% or higher.

This assumes the battery is actually fully charged when it's stated as being 99% or higher (mine only seems to go up to 99).  Maybe this is what's going on internally anyway, but if not it might give a more accurate SoC reading than the one stated, since there's talk of that being difficult to read and therefore sometimes inaccurate.

Does anyone have any thoughts?  Or the ability to write the necessary code?!  I think it might benefit those of us who are seeing the odd SoC drop.

**Reply by C00K50N** [reply]
I've cobbled something together that's not quite as complex as I had in mind above, but will hopefully serve the purpose at least temporarily...

My lack of ability meant I couldn't work out how to start something counting at [max capacity] and then monitor changes to then subtract discharge_sum and add charge_sum.

I did, however, figure I could manually check the value history of bat_discharge_sum and bat_charge_sum the last time I know the battery was at full charge.  And, knowing my actual capacity in kWh, I then wrote the following to essentially make that known 100% point a value of 15.6 (my installation's capacity) and then do the subtraction / addition from that point before making it a percentage value.

This seems to be working for now.  My code is below in case it's of use or interest to anyone else (check the notes below it too).  I'm intrigued to see the difference between this value and the stated SoC - it's out by 10-15%...  Right now by my calculation I'm at 35% and the battery_soc says I'm at 40%.  Assuming the charge_sum and discharge_sum figures are correct, this would probably account for the 'recalibration' others have been told the system is doing when we see the mysterious drop.

```
  - sensor:
      - name: "SoC Calculated"
        unit_of_measurement: "%"
        state_class: measurement
        state: >
          {{ ([A] - (states('sensor.bat_charge_sum')|float - [B]) - (states('sensor.bat_discharge_sum')|float - [C])) / [A] * 100 }}
```
[A - total capacity of batteries]
[B - value of sensor.bat_charge_sum at a point when the battery was known to be full]
[C - value of sensor.bat_discharge_sum at a point when the battery was known to be full]

I'm going to chart my calculated figure against the stated one, and we'll see how they compare over time.

**Reply by C00K50N** [reply]
Scratch that!  My maths is wrong.  Tweaking it now...

**Reply by C00K50N** [reply]
The correct line should be...

`{{ (([A] + (states('sensor.bat_charge_sum')|float - states('sensor.bat_discharge_sum')|float) - [B]) / [A] * 100) }}`

[A - total capacity of batteries]
[B - measured value of bat_charge_sum minus bat_discharge_sum at a point when the battery was known to be full^]

^ I did this today when the sun charged the battery to full before exporting - with its slow charge I have a feeling that will be a more reliable figure than when it's charged quickly from the grid.

This calculation assumes the charge_sum figures are updated accurately, but if they are it will be interesting to see how different this charts compared to the stated SoC figure.

**Reply by Thebig24fan** [reply]
Yer  I’ve had this problem too had an engineer sent by fox and said my system was acting abnormal and will look to replace inverter don’t know if that will fix it as think might have bms box problems as have this fault anyone else with the modbus installed seeing odd  amounts changed compared with Soc added ? We have a 13 kWh fox hv 2600 battery setup and like original poster now solar is getting lower starting to have to go low on some occasions  sometimes it does the protect chargers at 18-19% others at 14% 🧐 but been monitoring it  over last few days since had modbus setup  but was same odd Soc and added kWh before using inverter display info too, one night got down to 20 %  displayed soc so to change to 99% due to next rainy day that 79% added that should be 10.1+ kWh grid use to add this amount at least if you account for losses heat etc but we only put 7 kWh into the battery and it was at 99% for about 30 mins so should have done battery balancing as done this 5-6 times in last 2 weeks so pack should be fully balanced as have gone down as gone down as low as pack would go  then upto  99% charging with about 2 hours at 99% to balance the pack on all times with still hit and miss Soc before protecting charging happens, the other night got down to 14% lowest been this winter and changed  upto 99% again due to poor weather next day , so added 85% but only changed  8ish kWhs not 10.8 kWh + losses etc  I’ve check this import amount against smart meter data so it about right , any ideas? I’ve done fully system shutdown and  5-6 full charge nights and even been slow charging on solar till exporting and still hit and miss .  I’ll watch with interest. 

If you want an extra sample of data  tell we’re to put your code in ha and if possible add bits I need for 13 /12.8 nominal fox hv2600 setup thanks

**Reply by spmaid** [reply]
I'd recommend contacting FoxESS support, service@fox-ess.com and detail your issues and push for the BMS/battery firmware update.  I've been fine since 1.09, you can check your battery firmware on the inverter.  If you don't hear from service, maybe try directly with david.gu@fox-ess.com as he has been helping me since July.

**Comment by spmaid** [comment]
> Sorry for posting this here, but I don't use Facebook and so don't have access to the group on there. Yesterday there was mention about the issue with 'lost' SoC overnight, and also the forced charging issue (which is the self protect kicking in). I followed up with FoxESS support yesterday and they got back overnight to confirm they do now see the issue and expect to have a firmware fix coming soon, hopefully by end of the month or early next.
> 
> I am still questioning with them why the self protect charging is kicking in (for me) at 19%. They are saying sometimes the SoC might not be accurate etc. but I have challenged them on that as I don't see that is acceptable that I can no longer risk going below 20% SoC, it used to happily go down to 10% and as we get less and less solar now, that extra 10% will be very much needed.
> 
> Happy to post here again when I hear any more.

Yesterday I was offered an updated firmware, well, they asked if they could do it, I said sure, I noticed my power start pulling from the grid and not battery twice, so presume that was the firmware update but I haven't heard back from them since though. 

Not sure if this is the FW version info or not, but I am showing: 

Inverter Hardware Version : -Master : 1.48 Slave : 1.02 Manager : 1.45

Trouble is, with the lessening Solar, I'm charging each night, so I might not see the lost SoC anyway, but if we get a sunny day soon, I should be able to test it out.

**Reply by spmaid** [reply]
Noticed I now have the following, Manager has gone up to 1.47, although still no confirmation as to if I've been upgraded
 yet
Master : 1.48 Slave : 1.02 Manager : 1.47

**Reply by spmaid** [reply]
So this FW update has messed it right up, last night the battery was flat by 20:30, it was full due to good sunshine so should have lasted the night, I did a full charge during off peak but it only put in 6kWh and went from 20% SoC to 99% SoC inside an hour (not possible!) then after the off peak period ended it lost 25% in less than an hour and a half, then another 25% over the following hour, then at 07:10 it went from 44% to 13% SoC in 10 minutes.  I suspect that now it is the weekend I'll get no support, so as you can imagine, I am not pleased!  I'll be pulling from the grid all day.  The only positive is, I set the min SoC back down to 10% to see if it tried to emergency charge when it got to 19% like it has in the past, and it didn't, so that's something.

**Reply by spmaid** [reply]
an example of how messed up it is, the additional load during the 00:30 to 04:30 is my car charging.  You'll see only about an hour of charge going in to the batteries, I have a 6kW inverter and 18.2kWh of batteries, so the sums don't add up!

<img width="1129" alt="image" src="https://user-images.githubusercontent.com/99452615/198818973-ceac9543-3697-40cb-824e-4dbe131c3aa8.png">

**Reply by C00K50N** [reply]
That's most annoying!  I hope you get a response soon from support.  I hope they can fix it too!

**Reply by spmaid** [reply]
Overnight, it was even worse, charging for just 20 minutes getting to 99% SoC, and so, I already have a flat battery. 

<img width="1098" alt="image" src="https://user-images.githubusercontent.com/99452615/198865242-8e7cdad5-3831-48ce-bf6b-e84367dbe5be.png">

**Reply by spmaid** [reply]
Sorry to post again, just had to share how epically bad this has gone!  BMS reporting it dropped to 0% SoC and the the SOC jumping around all day as and when the sun came out.  Restarted the Inverter and BMS and it has come back with 25% SoC now, so I am praying they batteries did not go totally flat.  Also hoping that support with be back in the office overnight, guessing they are about 8 hours ahead.  

<img width="1070" alt="image" src="https://user-images.githubusercontent.com/99452615/198897443-2c36990f-2177-4282-b57c-f0dde45334b9.png">

**Reply by Daibutt** [reply]
Wow, that is pretty abysmal... hope you get some joy out of FoxESS. 
I do get the "random" SoC drops and increases (mostly drops) but usually around 20-25% drop. 
Energy balance on system (i.e ins - outs) usually stacks up roughly so it's not losing energy, but as SoC governs what I think I want to charge & what the inverter lets me charge, it's pretty integral to managing the system. 
Yours as shown above is impossible to manage (or even believe!).  Is this with HV2600s, cubes or LV batteries? 
BMS issue I guess, can't see the batteries all suddenly deciding they have no charge, and I assume the inverter just reports what the BMS tells it. 
Good luck, let us know what Fox say/do !

**Reply by spmaid** [reply]
Sorry for the delayed reply, I was away for a few day.  So, we're back working again now, FoxESS support didnt get back to me until late Monday, and wanted to run it for one more day so they could grab data.  They rolled back the FW I guess.  And as you say, BMS must have been lying as the batteries can't do what the graphs show!  I have 7xHV2600s. 

Would love to know what it is that went wrong, and I'll make sure I never let them do a FW upgrade on a Friday! I should know better, we're not allowed to do major changes on a Friday, and even when we do do changes, we normally keep an eye on it afters!

**Comment by C00K50N** [comment]
I've  been monitoring for 3 days now, and the stated SoC does seem to stray away from my calculated one.  I started my calculated sensor at 100% on a sunny day, when the batteries filled up to a stated 99% then kept charging a bit more (didn't hit a stated 100%, but that's OK).  You can see just after that point to the far left on the chart below, where my calculated SoC started higher than the stated value.  

Since then it has strayed away, starting above but mostly below the stated value.  It's not too far adrift, and in fact is pretty much bang on at the point of writing.  Like you, @spmaid we're also charging every night so I haven't seen the random drop in SoC.  I'll leave this Calculated SoC sensor charting though, as I'll be interested to see how it compares next time we do see the drop.

![image](https://user-images.githubusercontent.com/102669305/198675577-e2032c8c-81f4-43dd-9b5b-91697aa05c95.png)

**Reply by C00K50N** [reply]
After a charge to 'full' last night, it has drifted a bit further from my calculated value...  The stated value does seem to jump faster as it gets closer to 99% - I know others see this too...

![image](https://user-images.githubusercontent.com/102669305/198821518-00fe8b1c-c466-450d-ba05-e359448ca02b.png)

**Comment by spmaid** [comment]
Just thought I'd post an update as it's been a while since I reported all the issues with losing SoC etc.  I think I finally now have a stable firmware on the batteries that has stopped the lost SoC.  I got updated to 1.9 (on the inverter menu under battery version it says 01.00.01.09 for battery 1 02.00.01.09 for battery 2 etc.). I now have what I consider to be stable and predictable SoC changes.  Before, I estimate I was only getting about 40% of my total storage as useable, but now the amount going in and out equates to expected SoC changes.  

I have been dealing with FoxESS Support service@fox-ess.com but also directly with david.gu@fox-ess.com I am not sure if they plan to eventually automatically roll this out or not but it might be worth dropping them an email if you are suffering.  

To clarify, I have a H1-6.0-E but the firmware that has been updated is for the batteries/BMS.  I have HV2600s, some Ver C. and Some Ver D. 

[Edited to clarify version number and to add Inverter and battery details.]

**Reply by Thebig24fan** [reply]
Thanks I’ll shot him a email as still having  hit and miss bottom Soc and incorrect energy added  the engineer said might be bms or inverter . 
got 13 kWh hv2600 battery and h1 5.0 inverter on 
Master 1.48 slave 1.02  manger 1.50  like look on invter tomorrow for battery version

**Comment by tagware** [comment]
This is why I complained about End users!

What Fox ESS Model, system? Quoting versions is wasting time without these
items included.

Regards
*David Vincent*
*Mobile:* *07785 114965*

*Website: www.tagware.co.uk <http://www.tagware.co.uk/> *
*Blog:     **blog.tagware.co.uk* <http://tagware.info/>

<http://www.tagware.co.uk/>   *13 Sandersfield Road, **Banstead, **Surrey,**SM7
2DH. * *Registered No: 03006925 England. ** - *   *wacky.battle.blaze
<https://map.what3words.com/wacky.battle.blaze> *

This e-mail contains confidential information and is for the exclusive use
of the addressee's. If you are not the addressee, then any distribution,
copying or use of this e-mail is prohibited. If received in error, please
advise the sender and delete it immediately. We accept no liability for any
loss or damage suffered by any person arising from the use of this e-mail.
How we use your data can be viewed here
<https://tagware.co.uk/privacy-policy/>

**Reply by spmaid** [reply]
Not sure if this was aimed at me, but I have updated my post to add inverter and battery details and clarify the firmware version.  Apologies for missing this out.

**Comment by tagware** [comment]
I'm sorry. I didn't mean the response to be rude. I apologies if it came
across like that.

I worked in 4th line IT 20+ years and by the time the problem got to me
every minute was counting. I used to get my own spot on "News at Ten" when
the systems I looked after had a bad day. :o)

Hence, the direct email approach to 2nd/3rd line engineers, that normal
have cocked up. The few words and directness normally means they didn't
cock it up again..

On a more positive note. It will also help others if they searching Google
or the forum to have the model numbers included.  As you said yourself, you
took 12+ hours before your question was answered.

Cheers.

Regards
*David Vincent*
*Mobile:* *07785 114965*

*Website: www.tagware.co.uk <http://www.tagware.co.uk/> *
*Blog:     **blog.tagware.co.uk* <http://tagware.info/>

<http://www.tagware.co.uk/>   *13 Sandersfield Road, **Banstead, **Surrey,**SM7
2DH. * *Registered No: 03006925 England. ** - *   *wacky.battle.blaze
<https://map.what3words.com/wacky.battle.blaze> *

This e-mail contains confidential information and is for the exclusive use
of the addressee's. If you are not the addressee, then any distribution,
copying or use of this e-mail is prohibited. If received in error, please
advise the sender and delete it immediately. We accept no liability for any
loss or damage suffered by any person arising from the use of this e-mail.
How we use your data can be viewed here
<https://tagware.co.uk/privacy-policy/>

**Comment by Thebig24fan** [comment]
Thanks for the email address I’ve emailed him as not my response from fox support

From the desk of Nickie
________________________________

**Reply by Thebig24fan** [reply]
Update after a lot to and frow  they did the update to our system and it’s now giving more out the battery and charging more most of the time 🧐 before was getting  6 ish kWh out the battery before it would import at 18-20% on the Soc now it’s mostly giving 10-11 kWh at 10-11 %  before it imports like it used too but on 3 occasions it’s only give 6.5 -7 kWh after 100% charging and  dropped  unexpectedly  fast and for no reason and started importing  when there was little draw etc  but mostly it’s now much improved  so thanks for your help and the email  thankfully they read emails unlike uk support and did he the update after I showed photos proof as for a start he said if looked and your system looks to be within spec and no problems he could see etc . So if you have problems take screenshots etc and don’t take no for answer!

**Comment by spmaid** [comment]
Thought I'd post an update to this, as I think the issue has finally been found!  1 of the batteries appears to be faulty, I have removed it from the stack, and all has been well since, well maybe other than not as much going in/out as I'd expect, but that might be temperature related at the moment, I'll see if it improves as the weather warms.  Now just waiting on an RMA replacement battery.

Edit: During the last few weeks before this issue was found, I was only getting about 60% of useable power out of the batteries with massive jumps in SoC during both charge and discharge.  All is now quite stable, with it getting to about 95%, holding for a bit on a lower charge, then getting to 100%.  Fingers crossed for no further issues!
